### ISSUE TITLE
Document how to contribute by doing triage

### ISSUE THREAD

**Issue by eileenmcnaughton** [civicrm/civicrm-dev-docs#470]
@ErikHommel has offered to help up with doing triage. I think that since this is a task community members take on a times to contribute we should document it. Here is my take:

Triage is doing one or more of the following actions, either to issues you logged or just randomly from the list below. Ideally we would co-opt 2-3 people to spend an hour or 2 a week on triage.

The first priority with triage is to ensure incoming issues [Incoming issues](https://issues.civicrm.org/jira/issues/?filter=32110&jql=project%20in%20(CRM%2C%20INFRA)%20AND%20resolution%20%3D%20Unresolved%20AND%20created%20%3E%3D%20-7d%20ORDER%20BY%20priority%2C%20key)  are triaged 

1. determine if it is accurately categorised as bug vs improvement
2.  detemine if priority is appropriate
3. try to ensure component description & title are correct & meaningful 
3.  determine if the issue still is replicable on the latest master
4. ask any additional questions that seem necessary
5. provide any first-level support you feel able to. 
6. announce any new regressions or critical issues (that really are) on the product-maintenance channel in chat
7. discuss any that are hard to triage on the product-maintenance channel in chat

This is a good example of a triager doing the above steps - https://issues.civicrm.org/jira/browse/CRM-20711

https://issues.civicrm.org/jira/issues/?filter=-5&jql=resolution%20%3D%20Unresolved%20order%20by%20type%20ASC%2C%20priority%20DESC%2Cupdated%20DESC

Prioritisaion can be tricky. I consider 

**'Blocker'** to be only applicable to something that would block the next release going out or change the release schedule to get a new hot fix out.

**Critical** is something that - causes data integrity errors or a fatal error or for CiviCRM to do something substantially different to what it 'says it will do' - e.g the group of people who receive an email being different from those selected.

**Important** is generally for subsytems IMHO - ie. something that would be a blocker or critical for work that is not currently part of our core offering - such as Drupal 8, Mosaico, new CiviCase

**Major** includes things that have an notable impact on many users or significant impact on a smaller set.

**Minor** is a significant impact in an obscure scenario or a minor impact or an important improvement request.

**Trivial** is ... trivial - e.g wording, translation (sorry people). most improvement requests

Where an issue has been logged or championed by someone who could fix it themselves (e.g a partner) but has not I think that is an indication to err on the side of lowering the priority or considering it to be an 'Improvement request' rather than a bug.  Ditto where a bug is newly discovered but has been around for a long time. For example if a behaviour has been in place since 4.5 then it would be hard to make a case for it to be a major bug.

**Comment by JoeMurray** [comment]
The formal definitions and levels of important for Civi priorities in JIRA are:
Blocker	Blocks development and/or testing work, production could not run.	
Critical	Fatal errors, loss of data, recent regression or bug that could affect the reputation of a site (e.g sending out reminder to the wrong people).	
Important	Important to the CiviCRM product's future & reputation	
Major	Major loss of function.	
Minor	Minor loss of function, or other problem where easy workaround is present.	
Trivial (Default)	Cosmetic problem like misspelt words or misaligned text.

So Important does not have to necessarily be about a subsystem, it can be anything that is important to a) future (eg supporting Doctrine, supporting angularJS, integrating with D8), or b) our reputation (eg making product WCAG accessible).

**Comment by ErikHommel** [comment]
I think the important definition from JIRA is way too ambiguous/subjective. It can work for closed source or a single supplier but in a community something like reputation is subject to all kinds of individual viewpoints or priorities.
Can we come up with a better one? I will have a try too.

**Comment by ErikHommel** [comment]
I like the major definition as "Major includes things that have an notable impact on many users or significant impact on a smaller set." because I can apply it with an acceptable amount of interpretation. If I try to get a level above that but not critical yet I reckon it would be:
"Important includes things that have a significant impact on many users but a workaround is still possible". After all, if it would be critical to a reputation that would damage my business I would fix it even if the priority was trivial.
What do you think?

**Comment by JoeMurray** [comment]
No, that's too bug oriented. This was added if I'm not mistaken to identify big initiatives that will help advance the community in significant ways. Responsive design. Technology replacement where existing deprecated stuff is holding developers back, etc. To me, not being illegal to use in the US, Australia, Canada and elsewhere due to not meeting WCAG rules is a huge reputation issue.

**Comment by JoeMurray** [comment]
https://www.drupal.org/about/features/accessibility
https://make.wordpress.org/accessibility/2016/03/21/wordpress-goes-wcag/
https://www.joomla.org/accessibility-statement.html

**Comment by eileenmcnaughton** [comment]
I think that's the issue I have with important - I'm happy for it to be a feature request definition or subsystem but I feel like for core bugs it's better left out & blocker, critical, major.... being the categories

**Comment by eileenmcnaughton** [comment]
I guess to clarify - I think that there should be some difference between priorities for bugs & for proposed improvements - the categorys for bugs make sense but when you apply them to improvements they dont' really. In terms of improvements I think you really have 3 levels

- Important - these matter enough that they *might* be a candidate for core or other pro bono time because they are important for CiviCRM's future etc. We should look through these to see if any are GSOCable
- Somewhat important - generally important to some users (who might care enough to sponsor them)
- Unimportant - not really important (but they might still get done because they are trivial or they scratch someone's itch

**Comment by eileenmcnaughton** [comment]
If I'm being honest the way I currently treat important is as a dumping ground for work that I think should really be sponsored, is not really a bug appropriate to people doing voluntary or community support or is something that some is trying to sneak onto someone else's to do list because 'hey man it's really really important... oh no I'm far to busy to do it myself' - so basically I categorise things to important to humour people & then totally ignore any bugs that are set to 'important'. 

If I'm looking to address bugs I currently look at critical & blocker. We are getting close enough to having those fixed I've started to turn my eye to major bugs but I don't look at important.

**Comment by ErikHommel** [comment]
The only problem I have with the reputation one is that IMHO it is far too subjective and depending on business opportunities. I think as a community that is asking for difficult and heated debate so I would love to leave something like that out of the definition. If @eileenmcnaughton is ignoring important and @JoeMurray thinks it is vital to his business I guess I can safely say we have a definition that is not working.

**Comment by ErikHommel** [comment]
Again @JoeMurray my problem is not with WCAG or the contents of this issue, my problem is that I still do not have a clue when to use important. I certainly do not use it if I want @eileenmcnaughton to look at the issue :-)

**Comment by JoeMurray** [comment]
I like not using important for bug tracking issues. Why don't you just use it for issues marked as improvements. 

With respect to whether 'reputation' can be turned into something intersubjective and not purely subjective, and thus be useful for prioritzing improvements, I'd say we can likely find some 3rd party sources for such things. For example, what criteria are used by CRM software ranking sites and surveys. And note a) reputation is only one part of description of priority that would be contrasted with somewhat important and not important, and b) there can be reasonable agreements on some other aspects of issue/bug prioritization.

**Comment by eileenmcnaughton** [comment]
I'm happy with the idea of important as a priority for improvements - tbh I'd probably prefer we had completely different sets of priorities for bugs vs improvements since it doesn't make sense to measure them on the same yardstick.

One thing I'm thinking is that when I talk about important for subsystems that's not exactly right - we have a situation where we get bugs against improvements that are not fully stable - notably drupal 8 & next generation case management. In both these cases I think they are outside our normal bug prioritisation & that is what I mean by subsystems.

**Comment by ErikHommel** [comment]
I am happy with important only for improvements. Whenever we run into an improvement where I am not sure about the importance I will give a shout in the Mattermost room.

**Comment by eileenmcnaughton** [comment]
OK cool - I recently downgraded a 'critical' to 'important' because it only manifests when using CiviRules :-)

**Comment by eileenmcnaughton** [comment]
probably superceded by #531
